riscv_config/schemaValidator.py

```diff
@@ -391,3 +391,14 @@ def _check_with_sedeleg_reset(self, field, value):
         s = format(value, '#{}b'.format(supported_xlen[0] + 2))
         if (s[-11:-8]) != '000' and value >= int("400", 16):
             self._error(field, " 11,10,9 bits should be hardwired to 0")
+
+    def _check_with_p_check(self, field, value):
+        check = False
+        if 'implemented' in value:
+            if value['implemented']:
+                check = True
+
+        if not 'Zpn' in self.document['ISA'] : #Zpn is mandatory for P-SIMD
+            self._error(field, "Zpn is not present")
+        elif not check:
```

Comment:
It is legal for the `implemented` field is `false` if the `P` extension is not present. So the check should account for this and throw an error if it is implemented without P extension or P extension is present and implemented is false.